src/test/java/org/gridsuite/shortcircuit/server/ParametersControllerTest.java

```diff
@@ -0,0 +1,162 @@
+import org.gridsuite.shortcircuit.server.service.ShortCircuitService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultMatcher;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith({ MockitoExtension.class })
+@WebMvcTest(controllers = { ParametersController.class })
+class ParametersControllerTest implements WithAssertions {
+    private static final String DEFAULT_PARAMETERS_JSON = "{\"predefinedParameters\":null,\"parameters\":null,\"cei909VoltageRanges\":[{\"range\":{\"comparator\":\"INSTANCE\",\"maximum\":199.99,\"minimum\":10.0,\"naturalOrdering\":true},\"rangeCoefficient\":1.1,\"voltage\":\"NaN\",\"minimumNominalVoltage\":10.0,\"maximumNominalVoltage\":199.99},{\"range\":{\"comparator\":\"INSTANCE\",\"maximum\":299.99,\"minimum\":200.0,\"naturalOrdering\":true},\"rangeCoefficient\":1.09,\"voltage\":\"NaN\",\"minimumNominalVoltage\":200.0,\"maximumNominalVoltage\":299.99},{\"range\":{\"comparator\":\"INSTANCE\",\"maximum\":500.0,\"minimum\":300.0,\"naturalOrdering\":true},\"rangeCoefficient\":1.05,\"voltage\":\"NaN\",\"minimumNominalVoltage\":300.0,\"maximumNominalVoltage\":500.0}]}";
+    private static final MediaType TEXT_PLAIN_UTF8 = new MediaType("text", "plain", StandardCharsets.UTF_8);
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private ShortCircuitService shortCircuitService;
+
+    @AfterEach
+    void checkMocks() {
+        Mockito.verifyNoMoreInteractions(shortCircuitService);
+    }
+
+    @Test
+    void testGetExistingParameters() throws Exception {
+        final UUID arg = UUID.randomUUID();
+        final Optional<ShortCircuitParametersInfos> returned = Optional.of(new ShortCircuitParametersInfos(null, null));
+        when(shortCircuitService.getParameters(any(UUID.class))).thenReturn(returned);
```

Comment:
if you put eq(arg) instead of any(UUID.class) it ensures that your service has been called with the right argument then you don't have to add your check with the capture

Reply:
that is not the same thing, I mock the function to return something when called to not have a `NullPointerException` (which need to be debugged else to know if the cause is the mock or the code), then we check that it's the correct ID that was passed.